### DISCUSSION TITLE
use of the slice function with multiple dimensions

### DISCUSSION THREAD

**Discussion by VGuette** [pydata/xarray#6571] [Q&A]
Hi everyone. 
I'm using Xarray and netCDF meteorological data. I have the usual dimensions time, latitude and longitude and two main variables: the wind speed (time, lat, lon) and a latitudinal position (time, lon).
```
<xarray.Dataset>
    Dimensions:             (lon: 53, time: 25873, lat: 20)
    Coordinates:
      * lon                 (lon) float64 -80.0 -77.5 -75.0 -72.5 ... 45.0 47.5 50.0
      * time                (time) datetime64[ns] 1950-01-31 ... 2020-12-01
      * lat                 (lat) float32 70.0 67.5 65.0 62.5 ... 27.5 25.0 22.5
    Data variables:
        uwnd                (time, lat, lon) float32 -0.0625 0.375 ... -1.812 -2.75
        positions           (time, lon) float64 40.0 40.0 45.0 ... 70.0 70.0 70.0
```
For each (time, lon), I'd like to calculate a latitudinal average around the positions.
If I do a loop, I would do this (for a +-2.5° latitude average):

```
for i in ds.lon.values:
            for t in ds.time.values:
                  wind_averaged.loc[t,i]=ds.uwnd.sel(lon=i,time=t).sel(lat=slice(2.5+ds.positions.sel(lon=i,time=t).values,ds.positions.sel(lon=i,time=t).values-2.5)).mean('lat')
```

This is obviously very bad and I wanted to use `slice()` like this:
`wind_averaged=ds.uwnd.sel(lat=slice(2.5+ds.positions.values,ds.positions.values-2.5)).mean('lat')`

but it gives an error because I

` cannot use non-scalar arrays in a slice for xarray indexing`

Is there any alternative to do what I want without doing two for loops by using Xarray power?

Thanks

**Comment by dcherian** [comment]
Does ```.sel(lat=np.arange(2.5+ds.jet_positions.values,ds.jet_positions.values-2.5, 0.5), method="nearest")``` work? I chose `0.5` because that's the spacing on the `lat` variable. Here are the [docs](https://docs.xarray.dev/en/stable/user-guide/indexing.html#more-advanced-indexing) for this feature.

**Reply by VGuette** [reply]
Hi! 
Unfortunately no, I have:
`The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`
I guess this is because positions depends also on (time,lon), which is the main difficulty.

**Reply by dcherian** [reply]
Sorry we need to pass DataArrays so lets remove the `.values`. Try `.sel(lat=np.arange(2.5+ds.jet_positions,ds.jet_positions-2.5, 0.5), method="nearest")`

**Reply by VGuette** [reply]
Thanks for looking at this.
ds.jet_positions is now a DataArray like this:

```
<xarray.DataArray 'jet_positions' (time: 25873, lon: 144)>
array([[70. , 60. , 47.5, ..., 70. , 70. , 70. ],
       [35. , 35. , 35. , ..., 50. , 50. , 40. ],
       [35. , 35. , 35. , ..., 50. , 50. , 37.5],
       ...,
       [42.5, 42.5, 42.5, ..., 42.5, 42.5, 42.5],
       [40. , 40. , 40. , ..., 40. , 40. , 40. ],
       [37.5, 37.5, 37.5, ..., 40. , 40. , 37.5]])
Coordinates:
  * lon      (lon) float32 -180.0 -177.5 -175.0 -172.5 ... 172.5 175.0 177.5
  * time     (time) datetime64[ns] 1950-01-31 1950-02-01 ... 2020-12-01
```

But it still leads to the same error unfortunately!
`The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`
